### ISSUE TITLE
XQuery: Casting maps to functions

### ISSUE THREAD

**Issue by ChristianGruen** [BaseXdb/basex#1204]
The following cast raises `XPTY0004`:

```xquery
declare function local:f($x as function(xs:string) as xs:string) {
  $x
};
local:f(map { "a": "b" })
```

**Comment by ChristianGruen** [comment]
Related:

```xquery
map { 'a': 'b' } instance of function(xs:string) as xs:string?,
map { 'c': 'd' } instance of function(xs:string) as xs:string
```

**Comment by ChristianGruen** [comment]
Resolved (7cb8890cd4225529de38696cb370d484166c120f, 1c00bbca4b5f5e655fc287da3eacd30953a2eb49)
